CMakeLists.txt

```
@@ -79,6 +79,10 @@ if (MSVC)
         ${gvk_IncludeScope} "/MP")
 endif(MSVC)
 
+# Disable obsolete functionality in ImGui:
+# (Can be removed once ImGui has deprecated these for good, and acts as if these were defined.)
+add_definitions(-DIMGUI_DISABLE_OBSOLETE_KEYIO -DIMGUI_DISABLE_OBSOLETE_FUNCTIONS)
```

add_definitions is superseded by newer functions that are more explicit and don't require the "-D".
I'd suggest replacing with the following: 
```suggestion 
target_compile_definitions(${PROJECT_NAME} PUBLIC IMGUI_DISABLE_OBSOLETE_KEYIO IMGUI_DISABLE_OBSOLETE_FUNCTIONS)
```

For more information see [CMake Docs](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html?highlight=target_compile_definitions)